hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/auth/HmppsResourceServerConfiguration.kt

```diff
@@ -81,10 +81,18 @@ class HmppsReactiveResourceServerConfiguration {
     http {
       csrf { disable() }
       authorizeExchange {
-        customizer.unauthorizedRequestPathsCustomizer.unauthorizedRequestPaths.forEach { authorize(it, permitAll) }
-        customizer.anyRequestRoleCustomizer.defaultRole
-          ?.also { authorize(anyExchange, hasRole(it)) }
-          ?: also { authorize(anyExchange, authenticated) }
+        customizer.authorizeExchangeCustomizer.dsl
+          ?.also {
+            // override the entire authorizeExchange DSL
+            customizer.authorizeExchangeCustomizer.dsl!!.invoke(this)
```

Comment:
odd that you need `!!` here?

Reply:
I know, I thought that too 🤷‍♂️

Reply:
can you not use `it` here?

Reply:
Yes and it doesn't require a !! 😄 

However the code then reads `it.invoke(this)` 😆. But if I rename it then I can do `dsl.invoke(this)` which is much better. Thanks!